opentype/opentype.d.ts

```diff
@@ -0,0 +1,249 @@
+        xMax?: number;
+        xMin?: number;
+        yMax?: number;
+        yMin?: number;
+    }
+
+    class GlyphNames {
+        private names;
+        constructor(post: Post);
+        glyphIndexToName(gid: number): string;
+        nameToGlyphIndex(name: string): number;
+    }
+
+    class GlyphSet {
+        private font;
+        private glyphs;
+        constructor(font: Font, glyphs: Glyph[] | { (): Glyph; }[]);
+        get(index: number): Glyph;
+        length: number;
+        push(index: number, loader: { (): Glyph; }): void;
+    }
+
+    interface KerningPairs {
+        [pair: string]: number;
+    }
+
+    function load(url: string, callback: { (error: any, font?: Font): void; }): void;
+
+    function loadSync(url: string): Font;
+
+    interface LocalizedName {
+        [lang: string]: string;
+    }
+
+    interface Metrics {
+        leftSideBearing: number;
+        rightSideBearing?: number;
+        xMax: number;
+        xMin: number;
+        yMax: number;
+        yMin: number;
+    }
+
+    function parse(buffer: any): Font;
+
+    class Path {
+        private fill;
+        private stroke;
+        private strokeWidth;
+        constructor();
+        bezierCurveTo(x1: number, y1: number, x2: number, y2: number, x: number, y: number): void;
+        close: () => void;
+        closePath(): void;
+        commands: PathCommand[];
+        curveTo: (x1: number, y1: number, x2: number, y2: number, x: number, y: number) => void;
+        draw(ctx: CanvasRenderingContext2D): void;
+        extend(pathOrCommands: Path | PathCommand[]): void;
+        lineTo(x: number, y: number): void;
+        moveTo(x: number, y: number): void;
+        quadraticCurveTo(x1: number, y1: number, x: number, y: number): void;
+        quadTo: (x1: number, y1: number, x: number, y: number) => void;
+        toPathData(decimalPlaces: number): string;
+        toSVG(decimalPlaces: number): string;
+        unitsPerEm: number;
+    }
+
+    interface PathCommand {
+        type: string;
+        x?: number;
+        y?: number;
+        x1?: number;
+        y1?: number;
+        x2?: number;
+        y2?: number;
+    }
+
+    interface Point {
+        lastPointOfContour?: boolean;
+    }
+
+    interface Post {
+        glyphNameIndex?: number[];
+        isFixedPitch: number;
+        italicAngle: number;
+        maxMemType1: number;
+        minMemType1: number;
+        maxMemType42: number;
+        minMemType42: number;
+        names?: string[];
+        numberOfGlyphs?: number;
+        offset?: number[];
+        underlinePosition: number;
+        underlineThickness: number;
+        version: number;
+    }
+
+    interface RenderOptions {
+        kerning: boolean;
+    }
+
+    interface Table {
+        [propName: string]: any;
+        encode(): number[];
+        fields: Field[];
+        sizeOf(): number;
+        tables: Table[];
+        tableName: string;
+    }
+
+}
+
+declare var opentype: typeof opentypejs;
```

Comment:
this package is also available as amodule, i.e. using npm and require. so add a module declaation e.g.:

```ts
declare module "opentype.js" {
    export = opentype;
}
```